src/test/clojure/xtdb/operator/scan_test.clj

```diff
@@ -676,3 +679,57 @@
                             [:put-docs :docs {:xt/id id}]))
 
   (t/is (= 2000 (count (xt/q tu/*node* '(from :docs [{:xt/id id}]))))))
+
+(deftest test-recency-filtering
+  (tu/with-tmp-dirs #{node-dir}
+    (binding [c/*page-size* 1
+              c/*l1-file-size-rows* 256
+              c/*ignore-signal-block?* true]
+      (with-open [node (xtn/start-node {:log [:in-memory {:instant-src (tu/->mock-clock (tu/->instants :year))}]
+                                        :storage [:local {:path (io/file (.toFile node-dir) "objects")}]
+                                        :indexer {:rows-per-chunk 1}})]
+        ;; 2020 - 2025
+        (let [tx-key (last (for [i (range 6)]
+                             (xt/execute-tx node [[:put-docs :docs {:xt/id 1 :col i}]])))]
+
+
+          (tu/finish-chunk! node)
+          ;; compaction happens in 2025
+          (c/compact-all! node)
+
+          (let [->old-merge-tasks scan/->merge-tasks
+                merge-tsk-cnt (atom 0)
+                query-opts {:node node
+                            :basis {:at-tx tx-key
+                                    :current-time (:system-time tx-key)}}]
+            (with-redefs [scan/->merge-tasks (fn [& args]
```

Comment:
as discussed, investigate whether we can call `->merge-tasks` as a unit test here instead